modules/core/src/main/java/org/apache/ignite/internal/util/subscription/OrderedMergePublisher.java

```diff
@@ -410,4 +405,33 @@ public void cancel() {
             }
         }
     }
+
+    private static class ErrorChain {
+        private final Throwable error;
+        @Nullable
+        private final ErrorChain next;
+
+        private final AtomicBoolean built = new AtomicBoolean(false);
+
+        private ErrorChain(Throwable error, @Nullable ErrorChain next) {
+            this.error = error;
+            this.next = next;
+        }
+
+        Throwable buildThrowable() {
+            if (!built.compareAndSet(false, true)) {
+                // Already built, so error already contains all subsequent exceptions attached.
```

Comment:
it's a false. You can return  not fully constructed `error`, due to flag amended before the `error` will be constructed.

Reply:
Fixed